engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/transform/hydration/batch/NadelBatchHydrator.kt

```diff
@@ -35,31 +36,40 @@ internal class NadelBatchHydrator(
                     aliasHelper = state.aliasHelper,
                     parentNode = parentNode,
                 )
+
+                // Becomes Pair<JsonNode, Instruction?>
                 when {
                     instructions.isEmpty() -> null
                     instructions.size == 1 -> parentNode to instructions.single()
-                    // Becomes Pair<JsonNode, Instruction>
                     else -> parentNode to getHydrationInstruction(state, instructions, parentNode)
                 }
             }
-            // Becomes Map<Instruction, List<Pair<JsonNode, Instruction>>>
+            // Becomes Map<Instruction?, List<Pair<JsonNode, Instruction?>>>
             .groupBy { pair ->
                 pair.second
             }
-            // Changes List<Pair<JsonNode, Instruction>> to List<JsonNode>
+            // Changes List<Pair<JsonNode, Instruction?>> to List<JsonNode>
             .mapValues { pairs ->
                 pairs.value.map { pair -> pair.first }
             }
 
-        return parentNodesByInstruction.flatMap { (instruction, parentNodes) ->
-            if (instruction == null) parentNodes.map {
-                NadelResultInstruction.Set(
-                    it.resultPath + state.hydratedField.fieldName,
-                    null
-                )
+        val jobs: List<Deferred<List<NadelResultInstruction>>> = coroutineScope {
```

Comment:
This one is still needed, but doesn't have as much of an impact. We already parallelize the batches later down, this parallelizes it if there are different instructions per field e.g.

```
interface Test {
  something: Thing
}
type TestA implements Test {
  something: Thing @hydrated
}
type TestB implements Test {
  something: Thing @hydrated
}
```

Then this would speed that up.